spring-integration-jpa/src/main/java/org/springframework/integration/jpa/outbound/JpaOutboundGatewayFactoryBean.java

```diff
@@ -71,18 +70,11 @@
 
 	private long replyTimeout;
 
-	private volatile boolean requiresReply = false;
+	private boolean requiresReply = false;
 
-	private volatile String componentName;
+	private String componentName;
 
-	/**
-	 * Constructor taking an {@link JpaExecutor} that wraps all JPA Operations.
-	 *
-	 * @param jpaExecutor Must not be null
-	 *
-	 */
-	public JpaOutboundGatewayFactoryBean(JpaExecutor jpaExecutor) {
-		Assert.notNull(jpaExecutor, "jpaExecutor must not be null.");
+	public void setJpaExecutor(JpaExecutor jpaExecutor) {
```

Comment:
I don't think we can remove the ctor if we're going to cherry-pick the change.

The problem is solved by the parser not using the ctor.

Maybe deprecate it?